causalml/inference/meta/explainer.py

```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+import numpy as np
+from eli5.sklearn import PermutationImportance
+import shap
+import matplotlib.pyplot as plt
+from lightgbm import LGBMRegressor
+from copy import deepcopy
+
+VALID_METHODS = ('gini', 'permutation', 'shapley')
+
+
+class Explainer(object):
+    def __init__(self, method, control_name, X, tau, classes, model_tau=None,
+                 features=None, normalize=True, override_checks=False, r_learners=None):
+        self.method = method
+        self.control_name = control_name
+        self.X = X
+        self.tau = tau
+        self.classes = classes
+        self.model_tau = LGBMRegressor() if model_tau is None else model_tau
+        self.features = features
+        self.normalize = normalize
+        self.override_checks = override_checks
+        self.r_learners = r_learners
+
+        self.importance_catalog = {'gini': self.gini_importance, 'permutation': self.perm_importance}
```

Comment:
Since `importance_catalog` is not specific to a class instance created, let's move it out of the `Explainer` class.